### PR TITLE
[Doc] Update PyTorch Lightning Related Docs

### PR DESCRIPTION
**Pull request by woshiyyya**
## Why are these changes needed?

There are many existing docs that use the `trainable` function or the external Ray Lightning library for PTL integration. Since we already proposed the new `LightningTrainer`, we should update the docs to use LightningTrainer, and inform the users to check out our new examples.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [x] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by matthewdeng — APPROVED** [review]

File: python/ray/tune/integration/pytorch_lightning.py [hunk 1]
```diff
@@ -77,6 +77,14 @@ class TuneReportCallback(TuneCallback):
 
     Reports metrics to Ray Tune.
 
+    .. note::
```
Comment (richardliaw) [inline python/ray/tune/integration/pytorch_lightning.py:80]: I thought we were not deprecating the tune integrations?

Comment (woshiyyya) [reply python/ray/tune/integration/pytorch_lightning.py:80]: Oh I see. Probably we just recommend the users to use LightningTrainer?

Comment (richardliaw) [reply python/ray/tune/integration/pytorch_lightning.py:80]: Yeah we can recommend but I thought we were not deprecating

Comment (woshiyyya) [reply python/ray/tune/integration/pytorch_lightning.py:80]: Ok got it! I've updated the docstring!

File: doc/source/ray-more-libs/using-ray-with-pytorch-lightning.rst [hunk 2]
```diff
@@ -1,6 +1,13 @@
+:orphan:
```
Comment (richardliaw) [inline doc/source/ray-more-libs/using-ray-with-pytorch-lightning.rst:1]: Should we just delete most of the text on this page then, and just leave the note?

Comment (woshiyyya) [reply doc/source/ray-more-libs/using-ray-with-pytorch-lightning.rst:1]: Sorry I just missed this comment. So we can still keep this doc in the drop-down menu, but only with this note?

Comment (woshiyyya) [reply doc/source/ray-more-libs/using-ray-with-pytorch-lightning.rst:1]: Users can check out the docs in previous release if they are still using ray_lightning.

**Review by richardliaw — APPROVED** [review]

**Review by zhe-thoughts — APPROVED** [review]

**Pull request merged**